src/Sylius/Bundle/ReviewBundle/Updater/ReviewableAverageRatingUpdater.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ReviewBundle\Updater;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Sylius\Component\Review\Calculator\AverageRatingCalculatorInterface;
+use Sylius\Component\Review\Model\ReviewableInterface;
+use Sylius\Component\Review\Model\ReviewInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.p.zalewski@gmail.com>
+ */
+class ReviewableAverageRatingUpdater implements ReviewableAverageRatingUpdaterInterface
+{
+    /**
+     * @var AverageRatingCalculatorInterface
+     */
+    private $averageRatingCalculator;
+
+    /**
+     * @var ObjectManager
+     */
+    private $reviewSubjectManager;
+
+    /**
+     * @param AverageRatingCalculatorInterface $averageRatingCalculator
+     * @param ObjectManager                    $reviewSubjectManager
+     */
+    public function __construct(AverageRatingCalculatorInterface $averageRatingCalculator, ObjectManager $reviewSubjectManager)
+    {
+        $this->averageRatingCalculator = $averageRatingCalculator;
+        $this->reviewSubjectManager = $reviewSubjectManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(ReviewableInterface $reviewSubject)
+    {
+        $this->modifyReviewSubjectAverageRating($reviewSubject);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateFromReview(ReviewInterface $review)
```

Comment:
Let's remove this method and pass the subject properly through state machine arguments. I pointed that out in one of my first comments. This will also allow us to get rid of the method below and keep this simple logic in ``update()``, this is an overkill now.